tests/cli/commands/test_scheduler_command.py

```diff
@@ -87,3 +91,18 @@ def test_graceful_shutdown(self, executor, mock_process, mock_scheduler_job):
                 scheduler_command.scheduler(args)
             finally:
                 mock_process().terminate.assert_called()
+
+    @mock.patch.dict(os.environ, {'AIRFLOW__CORE__PARALLELISM': '35'}, clear=True)
+    @mock.patch("airflow.cli.commands.scheduler_command.SchedulerJob")
+    @mock.patch("airflow.cli.commands.scheduler_command.Process")
+    def test_scheduler_print_config(self, mock_process, mock_scheduler_job):
+        args = self.parser.parse_args(['scheduler'])
+        with conf_vars({("core", "sql_alchemy_conn_cmd"): 'echo hello'}):
+            with contextlib.redirect_stdout(io.StringIO()) as temp_stdout:
+                scheduler_command.scheduler(args)
+                output = temp_stdout.getvalue()
+
+                assert "sql_alchemy_conn = < hidden > [cmd]" in output
+                assert "max_active_tasks_per_dag = 16 [airflow.cfg]" in output
+                assert "parallelism = < hidden > [env var]" in output
```

Comment:
Hmm, why is parallelism hidden 🤔

Reply:
I have `display_sensitive=False` (default value is False) which hides all values from env, vars and command. I don't want to pass display_sensitive=True since that might dump sensitive values to logs in case of starting as daemon or systemd. I guess we can try filtering for sensitive values in the for loop similar to how "default" values are filtered out so that other values from env/cmd are shown. Thoughts?

https://github.com/apache/airflow/blob/dc75f5d8768c8a42df29c86beb519de282539e1f/airflow/configuration.py#L717-L719

Reply:
That was what I was fearing it might be. I guess let's not worry about it now for now. This does make this less comprehensive than I was initially imaging as it's not uncommon to use env vars for non-sensitive config as well.

Reply:
Ough. I did not know `display_sensitive` is "so sensitive" :) .

I thin what we could do though is to implement a new flag in the dict `guess_sensitive_data` (?) and do few things there:
* hide anything coming from secret (I think there is flag that just shows path rather than value for that already and it is on by default) 
* We already have the list of sensitive configs:
```
    sensitive_config_values = {
        ('core', 'sql_alchemy_conn'),
        ('core', 'fernet_key'),
        ('celery', 'broker_url'),
        ('celery', 'flower_basic_auth'),
        ('celery', 'result_backend'),
        ('atlas', 'password'),
        ('smtp', 'smtp_password'),
        ('webserver', 'secret_key'),
    }
```
* we can pass config through "redact()" - and we could extend the semantics of `sensitive_var_conn_names` to also cover key names of sensitive data (or we could even add another config `sensitive_configuration_names` where user could specify which of the config they want to hide).

Reply:
Thanks @potiuk , I would like to have this as a separate issue since it might need more discussion on what and how to filter out items.

Reply:
> Thanks @potiuk , I would like to have this as a separate issue since it might need more discussion on what and how to filter out items.

Sure. Works for me :)